components/bigcommerce/product-card.js

```diff
@@ -1,10 +1,26 @@
+import { useMemo } from 'react'
 import Link from 'next/link'
+import { useCart } from '../../lib/cart';
 import AddToCartButton from './add-to-cart-button';
 import ProductPrices from './product-prices';
 
 export default function ProductCard({
   product,
 }) {
+  const { data, error } = useCart()
+  const products = data?.products || []
+  const count = useMemo(
+    () => products.filter(p => p.id === product.entityId).length,
+    [products, product.entityId]
+  )
+
+  if (error) {
+    // Currently the error is not being handled, so log it
+    // NOTE: if you see an error here a lot, it's because after making a file change
+    // the session DB resets, so the `session-id` cookie will no longer have a match
```

Comment:
To explain it better, when the API endpoint for `/api/cart` refreshes its state, the "local DB" disappears, that of course won't be a problem for applications with real DBs :laughing: